### PR TITLE
fix header for slices for small devices

### PR DESCRIPTION
**Pull request by Heyrows**
## Context

[Linear ticket for fixing headers](https://linear.app/prismic/issue/SM-933/[design]-header-bar-breaks-on-small-screens)
[Linear ticket for fixing edit button](https://linear.app/prismic/issue/SM-938/[design]-the-edit-slice-name-button-isnt-well-placed)

## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

## [OPT] Preview

For 938
image  

Our release. 
nicolas-i-o8koK0E-oRg-unsplash

### CODE REVIEW

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx [hunk 1]
```diff
@@ -33,6 +33,11 @@ const Header: React.FC<{
 }> = ({ component, isTouched, variation, onSave, isLoading }) => {
   const router = useRouter();
   const [showVariationModal, setShowVariationModal] = useState(false);
+  const sliceNameRef = useRef<HTMLSpanElement>();
+  const isSliceNameOverflowing = Boolean(
```
Comment (danigo99) [inline packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx]: Is this used anywhere?

**Comment by danigo99** [comment]
Also, didn't we want to display a tooltip when the name is cut short and we hover over it? Thought I remember Benjamin mentioning it

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx (hunk 1 above)
Comment (Heyrows) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx]: oh my, forgot to remove this.

**Comment by Heyrows** [comment]
Yes, it's a specific requirement from him, I used the title property so far (looked like the design)

**Review by danigo99 — APPROVED** [review]

**Pull request merged**